microk8s-resources/actions/enable.kubeflow.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+set -eu
+
+source $SNAP/actions/common/utils.sh
+
+
+CONTROLLER="uk8s"
+CLOUD="microk8s"
+MODEL="kubeflow"
+
+
+function print_message () {
+
+cat << EOF
+
+Congratulations, Kubeflow is now available.
+Run \`microk8s.kubectl proxy\` to be able to access the dashboard at
+
+    http://localhost:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#!/overview?namespace=$MODEL
+
+The central dashboard is available at http://$1/
+To tear down Kubeflow and associated infrastructure, run this command:
+
+    microk8s.juju kill-controller $CONTROLLER --destroy-all-models --destroy-storage
+
+For more information, see documentation at:
+https://github.com/juju-solutions/bundle-kubeflow/blob/master/README.md
+
+EOF
+
+}
+
+
+function enable_addons () {
+  "$SNAP/microk8s-enable.wrapper" dns storage dashboard juju
+  for i in {1..5}
+  do
+    "$SNAP/microk8s-status.wrapper" --wait-ready --timeout 60 && break || sleep 10
+  done
+}
+
+
+function bootstrap_and_deploy () {
+  "$SNAP/microk8s-juju.wrapper" bootstrap $CLOUD $CONTROLLER
```

Comment:
Test the existence of the controller and reuse it or fail gracefully. The same for adding the model.

Reply:
+1
juju controllers
will print the current controllers that are running, on which they'll be 1 or none.
juju show-controller
will provide some yaml that could be useful to see the controller state, or
juju status
may be preferred simply to make a read only call to the controller to see if it's running